libs/libc/stdio/lib_libfilesem.c

```diff
@@ -60,7 +60,8 @@ void lib_sem_initialize(FAR struct file_struct *stream)
  * lib_take_semaphore
  ****************************************************************************/
 
-void lib_take_semaphore(FAR struct file_struct *stream)
+int lib_take_semaphore(FAR struct file_struct *stream,
```

Comment:
Maybe we can add `flag` param instead of `take_func` and pass `O_NONBLOCK` is case if we need to use `_SEM_TRYWAIT` instead of `_SEM_WAIT`?
@yamt what do you think?